File: app/src/main/java/com/hdu/newe/here/page/main/leavenoto/LeaveNotoFragment.java
package com.hdu.newe.here.page.main.leavenoto;


import android.os.Bundle;
import android.support.constraint.ConstraintLayout;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.hdu.newe.here.R;
import com.hdu.newe.here.interfaces.DataCallBack;
import com.hdu.newe.here.utils.DatePickerFragment;

import org.w3c.dom.Text;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

/**
 * A simple {@link Fragment} subclass.
 */
public class LeaveNotoFragment extends Fragment implements DataCallBack {

    @BindView(R.id.tv_stu_name)
    TextView mTvStuName;
    @BindView(R.id.tv_stu_number)
    TextView mTvStuNumber;
    @BindView(R.id.tv_stu_faculty)
    TextView mTvStuFaculty;
    @BindView(R.id.tv_stu_speciality)
    TextView mTvStuSpeciality;
    @BindView(R.id.view_account)
    ConstraintLayout mViewAccount;
    @BindView(R.id.view_noto)
    FrameLayout mViewNoto;
    @BindView(R.id.tv_start_time)
    TextView mTvStartTime;
    @BindView(R.id.tv_finish_time)
    TextView mTvFinishTime;
    @BindView(R.id.view_time)
    ConstraintLayout mViewTime;
    @BindView(R.id.view_submit)
    FrameLayout mViewSubmit;
    Unbinder unbinder;
    @BindView(R.id.toolbar)
    Toolbar mToolbar;

    public LeaveNotoFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_leave_noto, container, false);
        unbinder = ButterKnife.bind(this, view);
        ((AppCompatActivity) getActivity()).setSupportActionBar(mToolbar);
        mToolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ((AppCompatActivity) getActivity()).finish();
            }
        });
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }


    @OnClick({R.id.tv_start_time, R.id.tv_finish_time, R.id.view_submit})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.tv_start_time:
                //实例化对象
                DatePickerFragment datePickerFragment = new DatePickerFragment();
                //调用show方法弹出对话框
                // 第一个参数为FragmentManager对象
                // 第二个为调用该方法的fragment的标签
                datePickerFragment.show(getFragmentManager(), "date_picker");
                break;
            case R.id.tv_finish_time:
                break;
            case R.id.view_submit:
                break;
            default:
        }
    }

    /**
     * 实现DataCallBack的getData方法
     */
    @Override
    public void getData(String data) {
        //data即为fragment调用该函数传回的日期时间
        mTvStartTime.setText(data);
    }
}

File: app/src/main/java/com/hdu/newe/here/page/main/leavenoto/LeaveNotoFragment.kt
package com.hdu.newe.here.page.main.leavenoto

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.text.format.Time
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import com.hdu.newe.here.R
import com.hdu.newe.here.utils.AdvancedTimePicker

import kotlinx.android.synthetic.main.fragment_leave_noto.*
import java.text.SimpleDateFormat
import java.util.*
import java.util.regex.Pattern

/**
 * A simple [Fragment] subclass.
 */
class LeaveNotoFragment : Fragment(), View.OnClickListener {


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_leave_noto, container, false)
    }

    override fun onResume() {
        super.onResume()
        var calendar = Calendar.getInstance()
        var year = calendar.get(Calendar.YEAR)
        var month = (calendar.get(Calendar.MONTH) + 1)
        var dayOfMonth = calendar.get(Calendar.DAY_OF_MONTH)
        var hour = calendar.get(Calendar.HOUR_OF_DAY)like "%02d"
        var minute = calendar.get(Calendar.MINUTE) like "%02d"
        val st = "${year}年${month}月${dayOfMonth}日 ${hour}:$minute"

        //默认请假时间2小时
        val timezone = TimeZone.getTimeZone("GMT+10")
        calendar.timeZone = timezone
        year = calendar.get(Calendar.YEAR)
        month = (calendar.get(Calendar.MONTH) + 1)
        dayOfMonth = calendar.get(Calendar.DAY_OF_MONTH)
        hour = calendar.get(Calendar.HOUR_OF_DAY)like "%02d"
        minute = calendar.get(Calendar.MINUTE) like "%02d"
        val ft = "${year}年${month}月${dayOfMonth}日 ${hour}:$minute"
        startTime.text = st
        finishTime.text = ft
        startTime.setOnClickListener(this)
        finishTime.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.startTime -> {
                AdvancedTimePicker(context!!) { year, month, dayOfMonth, hour, minute ->
                    startTime.text = ("${year}年${month + 1}月${dayOfMonth}日$hour:$minute")
                    finishTime.text = ("${year}年${month + 1}月${dayOfMonth}日${hour + 2}:$minute")
                }.show()
            }
            R.id.finishTime -> {
                AdvancedTimePicker(context!!) { year, month, dayOfMonth, hour, minute ->
                    finishTime.text = ("${year}年${month + 1}月${dayOfMonth}日$hour:$minute")
                }.show()
            }
        }
    }


    infix fun Int.like(pattern: String  ):String{
        return String.format(pattern,this)
    }
}
